src/test/scala/com/target/data_validator/validator/StringLengthCheckSpec.scala

```diff
@@ -0,0 +1,403 @@
+            |  column: item
+            |  minValue: 0
+            |  maxValue: 100
+            |
+              """.stripMargin
+        val json = io.circe.yaml.parser.parse(yaml).right.getOrElse(Json.Null)
+        val sut = json.as[Array[ValidatorBase]]
+        assert(sut.isRight)
+        assert(sut.right.get contains StringLengthCheck(
+          "item",
+          Some(Json.fromInt(0)),
+          Some(Json.fromInt(100)) // scalastyle:ignore
+        ))
+      }
+
+      it("fromJson with min") {
+        import JsonDecoders.decodeChecks
+        val yaml =
+          """---
+            |- type: stringLengthCheck
+            |  column: item
+            |  minValue: 0
+            |
+              """.stripMargin
+        val json = io.circe.yaml.parser.parse(yaml).right.getOrElse(Json.Null)
+        val sut = json.as[Array[ValidatorBase]]
+        assert(sut.isRight)
+        assert(sut.right.get contains StringLengthCheck(
+          "item",
+          Some(Json.fromInt(0)),
+          None // scalastyle:ignore
+        ))
+      }
+
+      it("fromJson with max") {
+        import JsonDecoders.decodeChecks
+        val yaml =
+          """---
+            |- type: stringLengthCheck
+            |  column: item
+            |  maxValue: 100
+            |
+              """.stripMargin
+        val json = io.circe.yaml.parser.parse(yaml).right.getOrElse(Json.Null)
+        val sut = json.as[Array[ValidatorBase]]
+        assert(sut.isRight)
+        assert(sut.right.get contains StringLengthCheck(
+          "item",
+          None,
+          Some(Json.fromInt(100)) // scalastyle:ignore
+        ))
+      }
+
+      it("fromJson without min or max") {
+        import JsonDecoders.decodeChecks
+        val yaml =
+          """---
+            |- type: stringLengthCheck
+            |  column: item
+            |
+              """.stripMargin
+        val json = io.circe.yaml.parser.parse(yaml).right.getOrElse(Json.Null)
+        val sut = json.as[Array[ValidatorBase]]
+        assert(sut.isRight)
+        assert(sut.right.get contains StringLengthCheck(
+          "item",
+          None,
+          None // scalastyle:ignore
+        ))
+      }
+    }
+
+    describe("Basic String Length check") {
+
+      it ("String length check fails numErrorsToReport:1") {
+        val dict = new VarSubstitution
+        val df = mkDataFrame(spark, defData)
+        val sut = StringLengthCheck("item", Some(Json.fromInt(5)), Some(Json.fromInt(6))) // scalastyle:ignore
+        val config = ValidatorConfig(1, 1, None, detailedErrors = true, None,
+          None, List(ValidatorDataFrame(df, None, None, sut :: Nil)))
+        assert(!config.configCheck(spark, dict))
+        assert(config.quickChecks(spark, dict))
+        assert(sut.failed)
+        assert(sut.getEvents contains
+          ValidatorCheckEvent(failure = true, "StringLengthCheck on column 'item'", 4, 2))
+        assert(sut.getEvents contains
```

Comment:
@samratmitra-0812 @dougb I think we need to be careful here, and in other tests below.  

This assert is based on the original ordering of `defData`.  There is no guarantee that when converted to a dataframe, partitioned, and operated on that we will get results back in the same order.  Since there are two potential modes of failure here this result could very well be the other one (empty string).

https://issues.apache.org/jira/browse/SPARK-16207?focusedCommentId=15349509&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-15349509